explorer/service/src/main/java/org/radixdlt/explorer/metrics/MetricsProvider.java

```diff
@@ -178,25 +170,62 @@ private void calculateMetrics(Map<String, SystemInfo> info) {
         long progress = Math.round(aggregatedProgress / nodeCount);
 
         synchronized (calculationLock) {
-            calculationIterations++;
             long seconds = (System.currentTimeMillis() - testState.getStartTimestamp()) / 1000 + 1;
             assert seconds > 0;
             averageTps = Math.round(progress / seconds);
             peakTps = Math.max(peakTps, tps);
             calculatedMetrics = new Metrics(tps, progress, averageTps, peakTps);
+            dumpCurrentMetrics();
         }
 
+        subject.onNext(calculatedMetrics);
+    }
+
+    /**
+     * Resets the metrics dump file to only contain a single header line,
+     * if a path to it has been set.
+     */
+    private void resetDumpFile() {
         if (metricsDumpPath != null) {
             try {
-                String line = calculatedMetrics.toString();
+                byte[] data = Metrics.DATA_HEADLINE.getBytes(UTF_8);
+                Files.write(metricsDumpPath, data, CREATE, WRITE);
+            } catch (Exception e) {
+                LOGGER.info("Couldn't reset metrics dump file: " + metricsDumpPath, e);
+            }
+        }
+    }
+
+    /**
+     * Dumps the current metrics to a file, if a path to it has been set.
+     */
+    private void dumpCurrentMetrics() {
+        if (metricsDumpPath != null) {
+            String line = calculatedMetrics.toString();
+            try {
                 byte[] data = line.getBytes(UTF_8);
                 Files.write(metricsDumpPath, data, CREATE, WRITE, APPEND);
```

Comment:
Only a good idea if you know the maximum size that the file can grow to.

Reply:
Hm... I'm not sure I follow. The file is cleared every time the test starts (enters the `STARTED` state). From there the file size grows linearly during the execution time of the test. Each line has fixed size so I can make a coarse grained estimate of the terminal dump file size. But I'm not sure I see what to do with this information.

Reply:
Yup 3d2adaa is the logic I was looking for. Nice O(1) algo.